lib/ansible/modules/windows/win_product_facts.ps1

```diff
@@ -57,9 +57,59 @@ if (-not $product_key) {
     }
 }
 
+try {
+    $license_info = Get-CimInstance SoftwareLicensingProduct | Where-Object PartialProductKey
+
+    switch ($license_info.LicenseStatus){
+        0 {
+            $winlicense_status="Unlicensed"
+        }
+
+        1 {
+            $winlicense_status="Licensed"
+        }
+
+        2 {
+            $winlicense_status="OOBGrace"
+        }
+
+        3 {
+            $winlicense_status="OOTGrace"
+        }
+
+        4 {
+            $winlicense_status="NonGenuineGrace"
+        }
+
+        5 {
+            $winlicense_status="Notification"
+        }
+
+        6 {
+            $winlicense_status="ExtendedGrace"
+        }
+
+        default {
+            $winlicense_status="NA"
```

Comment:
I feel this null or unexpected values should be N/A not NA. NA could potentially be an acronym for something else whereas N/A is the standard I've seen for not applicable.

Reply:
I'd got for $null, rather than a string. Unless that string somehow is the value that is expected in this case.